compliance_checker/cf/cf.py

```diff
@@ -409,16 +409,42 @@ def check_naming_conventions(self, ds):
         :param netCDF4.Dataset ds: An open netCDF dataset
         :rtype: Result
         '''
-        fails = []
-        total = len(ds.variables)
+        ret_val = []
+        variable_naming = TestCtx(BaseCheck.MEDIUM, '§2.3 Naming Conventions for variables')
+        dimension_naming = TestCtx(BaseCheck.MEDIUM, '§2.3 Naming Conventions for dimensions')
+        attribute_naming = TestCtx(BaseCheck.MEDIUM, '§2.3 Naming Conventions for attributes')
 
-        rname = re.compile("[A-Za-z][A-Za-z0-9_]*")
+        rname = re.compile("^[A-Za-z][A-Za-z0-9_]*$")
```

Comment:
Without the `^` and `$` it won't capture spaces and whitespace as illegal characters.